File: ambeth/Ambeth.Cache/ambeth/template/EmbeddedTypeTemplate.cs
using De.Osthus.Ambeth.Model;
using System;

namespace De.Osthus.Ambeth.Template
{
    public class EmbeddedTypeTemplate
    {
        public Object GetRoot(IEmbeddedType embeddedObject)
        {
            Object parent = embeddedObject.Parent;
            while (parent is IEmbeddedType)
            {
                parent = ((IEmbeddedType)parent).Parent;
            }
            return parent;
        }
    }
}

File: jambeth/jAmbeth.Cache/src/de/osthus/ambeth/template/EmbeddedTypeTemplate.java
package de.osthus.ambeth.template;

import de.osthus.ambeth.log.ILogger;
import de.osthus.ambeth.log.LogInstance;
import de.osthus.ambeth.model.IEmbeddedType;

public class EmbeddedTypeTemplate
{
	@SuppressWarnings("unused")
	@LogInstance
	private ILogger log;

	public final Object getRoot(IEmbeddedType embeddedObject)
	{
		Object parent = embeddedObject.getParent();
		while (parent instanceof IEmbeddedType)
		{
			parent = ((IEmbeddedType) parent).getParent();
		}
		return parent;
	}
}
